### PR TITLE
Proper interfaces for Visyn Views

### PR DESCRIPTION
**Pull request by dvzacharycutler**
Adding interfaces for visyn views. Build failing until https://github.com/datavisyn/tdp_core/pull/672 gets merged into develop and then ordino-2.0. 

Open to feedback on the types in src/views/interfaces.ts, especially how to name this type which is specific to the ordino "main" view, which is currently just the ranking view.

### CODE REVIEW

File: src/views/CosmicProxyView.tsx [hunk 1]
```diff
@@ -2,21 +2,22 @@
...
+export function CosmicView({ parameters, onParametersChanged }: CosmicViewPluginType['props']) {
   useEffect(() => {
     if (!parameters) {
       onParametersChanged({ currentId: '' });
     }
   });
 
-  console.log(parameters);
-
   return <iframe className="w-100 h-100" src="https://cancer.sanger.ac.uk/cosmic" />;
```
Comment (thinkh) [inline src/views/CosmicProxyView.tsx:16]: Will you do the generalization and configuration of the proxy view in a follow-up PR?

File: src/views/interfaces.ts [hunk 2]
```diff
@@ -16,6 +17,19 @@ export type VisynRankingViewPluginType<
      * TODO:: Type to IReprovisynServerColumn when we merge that into tdp_core
      */
     dataDesc: IServerColumn[] | any[];
+
+    idFilter: string[];
+
+    onIdFilterChanged(idFilter: React.SetStateAction<string[]>): void;
```
[the remark this answers is not in the document]
Comment (thinkh) [reply src/views/interfaces.ts:23]: @dvzacharycutler could you please reflect the renaming of https://github.com/datavisyn/tdp_core/pull/672/commits/e4b6c6d782901d80d1a16d411e2353acef3680fe here? Afterwards we can merge this PR, too.

[the remark this answers is not in the document]
Comment (dvzacharycutler) [reply src/views/interfaces.ts]: Done!

**Comment by thinkh** [comment]
@dvzacharycutler This CircleCI build is still failing, because the `develop` branch of tdp_core is not merged into `ordino-2.0`.

**Pull request merged**